### PR TITLE
Feature/integration pytorch tabular

### LINKED ISSUE #950: Integrate backbones from PyTorch Tabular for tabular classification and regression tasks
## 🚀 Feature

Use the models from here: https://github.com/manujosephv/pytorch_tabular

TabNet would work as currently but also have the others as options

### PR DESCRIPTION
**Pull request by Actis92**
In this PR I have integrated pytorch-tabular inside lightning flash, unsing the Adapter like what is already been done with icevision and pytorch-forecasting. 
I have added some parameters inside TabularData in order to have all the info required by the different models implemented in pytorch tabular.
And I have added tests and cli for TabularRegressor

Fixes #950

### CODE REVIEW

File: flash/core/integrations/pytorch_tabular/adapter.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        cat_dims: list,
+        num_categorical_fields: int,
+        num_numerical_fields: int,
+        output_dim: int,
+        backbone: str,
+        backbone_kwargs: Optional[Dict[str, Any]] = None,
+        loss_fn: Optional[Callable] = None,
+        metrics: Optional[Union[torchmetrics.Metric, List[torchmetrics.Metric]]] = None,
+    ) -> Adapter:
+        # Remove the single row of data from the parameters to reconstruct the `time_series_dataset`
```
Comment (ethanwharris) [inline flash/core/integrations/pytorch_tabular/adapter.py:31]: I think this comment could be removed:
```suggestion
```

File: flash/core/integrations/pytorch_tabular/backbones.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
+import functools
+import os
+
+from omegaconf import DictConfig, OmegaConf
+from pytorch_tabular.config import ModelConfig, OptimizerConfig
```
Comment (ethanwharris) [inline flash/core/integrations/pytorch_tabular/backbones.py:5]: This should be moved to only import if `_PYTORCHTABULAR_AVAILABLE`

File: flash/tabular/classification/model.py [hunk 3]
```diff
@@ -59,82 +61,72 @@ class TabularClassifier(ClassificationTask):
...
         self,
-        num_features: int,
-        num_classes: int,
-        embedding_sizes: Optional[List[Tuple[int, int]]] = None,
+        embedding_sizes: list,
+        categorical_fields: list,
+        cat_dims: list,
+        num_categorical_fields: int,
+        num_numerical_fields: int,
+        output_dim: int,
```
Comment (ethanwharris) [inline flash/tabular/classification/model.py:73]: It seems like we could remove some arguments here. Given `num_features` and `embedding_sizes` you have:
```
num_categorical_fields = len(embedding_sizes)
num_numerical_fields = num_features - num_categorical_fields
```
Or at least `num_categorical_fields = len(categorical_fields)`

**Review by ethanwharris — COMMENTED** [review]
@Actis92 This is awesome!!! Really cool work, left some comments :smiley:

**Comment by Actis92** [comment]
Thank you @ethanwharris for your comments 😃 I have done the changes that you have suggested, and I have done a refactoring of TabularClassifier, TabularRegressor because I have noticed that before I didn't set correctly metrics, loss functions, etc ... because it is needed to be set in the model defined by pytorch-tabular.
Let me know if you have any other observation 😃
